presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -360,10 +360,7 @@ public FSDataInputStream open(Path path, int bufferSize)
     public FSDataOutputStream create(Path path, FsPermission permission, boolean overwrite, int bufferSize, short replication, long blockSize, Progressable progress)
             throws IOException
     {
-        if ((!overwrite) && exists(path)) {
-            throw new IOException("File already exists:" + path);
-        }
-
+        //HACK: If overwrite is false then we should check if the file already exists but doing so might break S3 read-after-write consistency so skipping it.
```

Comment:
I would add `TODO` here. Even previously this code was only best effort since file could have been created by other process after existence check.

nit: add space after `//`

Reply:
Updated it